Sources/Lithosphere/DiagnosticEngine.swift

```diff
@@ -18,6 +18,12 @@ public final class DiagnosticEngine {
   /// The current set of emitted diagnostics.
   private(set) public var diagnostics = [Diagnostic]()
 
+  /// The set of in-flight diagnostics for a particular transaction.
+  private var transactionDiagnostics = [UUID: [Diagnostic]]()
+
+  /// The current transaction.
+  private var activeTransaction: UUID?
```

Comment:
Is UUID the right type to use here?  An Int would suffice.

Reply:
Why static? It can be an ivar of DiagnosticEngine.

Reply:
Oh, silly me.  Still thinking in terms of global UIDs.